tests/PHPStan/Rules/Variables/UnsetRuleTest.php

```diff
@@ -60,12 +61,16 @@ public function testBug2752(): void
 
 	public function testBug4289(): void
 	{
-		$this->analyse([__DIR__ . '/data/bug-4289.php'], [
-			[
-				'Cannot unset Bug4289\BaseClass::$fields property which might get hooked in subclass.',
-				25,
-			],
-		]);
+		if (PHP_VERSION_ID < 80400) {
+			$this->analyse([__DIR__ . '/data/bug-4289.php'], []);
```

Comment:
We usually do this by assigning `$errors` to `[]` and then to something on-empty in an if branch. So that there's only a single `analyse` call in the body.